pkg/features/pilot/pilot.go

```diff
@@ -113,6 +113,9 @@ var (
 		}
 		return time.Second * time.Duration(duration)
 	}
+
+	// SDSEnabled indicates whether SDS is enabled globally.
+	SDSEnabled = os.Getenv("SDS_ENABLED") == "1"
```

Comment:
See above - not clear what it means and when would a user enable it.

Normally if a envoy connects and its environment is set with an SDS - it doesn't make any sense to return a config using files, that node will likely not even have the secrets mounted so it would fail.

Likewise if a node doesn't have a UDS socket - there is no point or value to send a config with SDS, regardless of what the default setting is.

Reply:
btw, I give a try of the other approach - "pilot decide whether to generate sdsconfig or secret mount based on metadata that envoy passed to pilot"; this approach seems not work straightforward right now - 

1.pod annotation does have sds uds socket info like below, however seems it's on pod level, we cannot see it from envoy annotation either from [cluster](https://github.com/istio/istio/blob/master/pilot/pkg/proxy/envoy/v2/cds.go#L73) or [listener](https://github.com/istio/istio/blob/master/pilot/pkg/networking/plugin/authn/authentication.go#L91).
```
sidecar.istio.io/status: '{"version":"cd1c9ab7aa926545a9bd20bf4051da27519575d07a594e7f26fdb94e1a9847d0","initContainers":["istio-init"],"containers":["istio-proxy"],"volumes":["istio-envoy","sds-uds-path","istio-token"],"imagePullSecrets":null}'
```
2. for ingress/engress gateway, we don't have such annotation that include info like sds socket or secret mount.

To solve this, we may need to add sdsenable/jwt path to sidecar/gateway env variables to be reflected in metadata, @costinm wdyt ?